nemoguardrails/embeddings/basic.py

```diff
@@ -45,26 +45,16 @@ class BasicEmbeddingsIndex(EmbeddingsIndex):
         max_batch_hold: The maximum time a batch is held before being processed
     """
 
-    embedding_model: str
-    embedding_engine: str
-    embedding_params: Dict[str, Any]
-    index: AnnoyIndex
-    embedding_size: int
-    cache_config: EmbeddingsCacheConfig
-    embeddings: List[List[float]]
-    search_threshold: float
-    use_batching: bool
-    max_batch_size: int
-    max_batch_hold: float
+    # Instance attributes are defined in __init__ and accessed via properties
 
     def __init__(
         self,
-        embedding_model=None,
-        embedding_engine=None,
-        embedding_params=None,
-        index=None,
-        cache_config: Union[EmbeddingsCacheConfig, Dict[str, Any]] = None,
-        search_threshold: float = None,
+        embedding_model: Optional[str] = None,
+        embedding_engine: Optional[str] = None,
+        embedding_params: Optional[Dict[str, Any]] = None,
+        index: Optional[AnnoyIndex] = None,
+        cache_config: Optional[Union[EmbeddingsCacheConfig, Dict[str, Any]]] = None,
+        search_threshold: Optional[float] = None,
```

Comment:
If we must set defaults then better to change Optionnal[str] to str and make defaults more visible.

```suggestion
        embedding_model: str = "sentence-transformers/all-MiniLM-L6-v2",
        embedding_engine: str = "SentenceTransformers",
        embedding_params: Optional[Dict[str, Any]] = None,
        index: Optional[AnnoyIndex] = None,
        cache_config: Optional[Union[EmbeddingsCacheConfig, Dict[str, Any]]] = None,
        search_threshold: float = float("inf"),
```

Reply:
Already addressed in #1383